src/server/detail/save_stages_controller.cc

```diff
@@ -335,7 +338,8 @@ GenericError SaveStagesController::BuildFullPath() {
   if (auto err = ValidateFilename(filename, use_dfs_format_); err)
     return err;
 
-  SubstituteFilenameTsPlaceholder(&filename, FormatTs(start_time_));
+  SubstituteFilenamePlaceholders(&filename, kTSFormat, kYearFormat, kMonthFormat, kDayFormat);
```

Comment:
I personally think that having the constants inline is easier to read (and also shorter) than defining them above.